### PR TITLE
PLT-8528 removed check that was prevent submit if names were unchanged

### PR DESCRIPTION
**Pull request by stephenkiers**
Hitting save on the change channel name header without changing the name did nothing. 

There was a specific check preventing it from saving. I removed that check, and it now does as expected

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-8528

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)

### CODE REVIEW

**Comment by crspeller** [comment]
@stephenkiers You have some eslint errors.

File: components/rename_channel_modal/rename_channel_modal.jsx [hunk 1]
```diff
@@ -189,7 +187,7 @@ export class RenameChannelModal extends React.PureComponent {
 
         this.setState(state);
 
-        if (state.invalid || (oldName === channel.name && oldDisplayName === channel.display_name)) {
```
Comment (hmhealey) [inline components/rename_channel_modal/rename_channel_modal.jsx]: If the names didn't change, can we exit early without even sending a request to the server?

Comment (stephenkiers) [reply components/rename_channel_modal/rename_channel_modal.jsx]: Great point.

Comment (stephenkiers) [reply components/rename_channel_modal/rename_channel_modal.jsx]: fixed

**Pull request merged**